### ISSUE TITLE
Emacs Flymake: diagnostics not cleared sometimes

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#398]
Repro:
1. C-x b RET (open \*scratch\*)
2. M-x js-mode RET
3. Go to the first line and insert `//`
4. Go to the second line and insert `//`

At this point, Flymake *should* report no issues, but it reports an unclosed regexp literal error.

5. On a new line, type `documents`
6. Hover the mouse cursor over the word 'documents' and observe:
![Screenshot from 2021-07-21 20-18-08](https://user-images.githubusercontent.com/48666/126585558-0363dad4-eb7b-4282-b320-842572f822e7.png)

**Comment by wgrr** [comment]
@strager i think this bug is related with state track of quick-lint-js, reported a new issue (#400) for the crash that this input is doing to quick-lint-js

**Comment by strager** [comment]
> @strager i think this bug is related with state track of quick-lint-js process, reported a new issue (#400) for the crash that this input is doing to quick-lint-js

Do you think the plugin gets confused when quick-lint-js crashes?

**Comment by wgrr** [comment]
> Do you think the plugin gets confused when quick-lint-js crashes?

Yes, I'm still investigating that, apparently if quick-lint-js crashes flymake is never calling us back so we don't have a chance to disable ourselves or update the current errors, but im not 100% sure about this, what im tho is we are never taking the :panic branch, even if we crash.
I'll bet it has some relationship with signals.
